src/services/groups.ts

```diff
@@ -39,3 +40,47 @@ export function generateSecret(wordlist: string[], length: number): string {
   }
   return mnemonicWords.join('-');
 }
+
+/**
+ * Executes a query to retrieve the members of a group.
+
+ * @param {PostgresJsDatabase<typeof db>} dbPool - The database connection pool.
+ * @param {string} groupId - The ID of the user.
+ */
+export async function getGroupMembers(dbPool: PostgresJsDatabase<typeof db>, groupId: string) {
+  const response = await dbPool.query.groups.findMany({
+    where: eq(db.groups.id, groupId),
+    with: {
+      usersToGroups: {
+        with: {
+          user: true,
+        },
+      },
+    },
+  });
+
+  // Extract user objects from each usersToGroups entry
+  const groupMembers = response.flatMap((group) =>
+    group.usersToGroups.map((usersToGroup) => {
+      const { telegram, createdAt, updatedAt, username, email, ...userWithoutSensitiveInfo } =
+        usersToGroup.user;
+      return userWithoutSensitiveInfo;
+    }),
+  );
+
+  // Clean response
+  const cleanResponse = response.map((group) => {
+    const {
+      usersToGroups,
+      description,
+      secret,
+      groupCategoryId,
+      createdAt,
+      updatedAt,
+      ...groupWithoutSensitiveInfo
+    } = group;
+    return groupWithoutSensitiveInfo;
+  });
+
+  return { group: cleanResponse, groupMembers };
```

Comment:
lets only return the group members and not the whole group. so it would look like:

return groupMembers

Reply:
okey